src/tapedfunction.jl

```diff
@@ -83,9 +74,12 @@ val(x::Box) = x.val
 val(x::GlobalRef) = getproperty(x.mod, x.name)
 val(x::QuoteNode) = eval(x)
 val(x::TapedFunction) = x.func
-box(x) = Box(x)
 box(x::Box) = x
-Base.show(io::IO, box::Box) = print(io, "Box(", box.val, ")")
+box(x) = Box(gensym(), x)
+box(id::Symbol, x) = Box(id, x)
+box_any(x) = Box{Any}(gensym(), x)
+box_any(id::Symbol, x) = Box{Any}(id, x)
```

Comment:
Couldn't you just define a single argument outer constructor?
```suggestion
Box(x) = Box(gensym(), x)
Box{T}(x) where {T} = Box{T}(gensym(), x)
```
?